packages/history/index.ts

```diff
@@ -1008,40 +715,29 @@ function clamp(n: number, lowerBound: number, upperBound: number) {
   return Math.min(Math.max(n, lowerBound), upperBound);
 }
 
-function promptBeforeUnload(event: BeforeUnloadEvent) {
-  // Cancel the event.
-  event.preventDefault();
-  // Chrome (and legacy IE) requires returnValue to be set.
-  event.returnValue = '';
-}
-
 type Events<F> = {
-  length: number;
   push: (fn: F) => () => void;
   call: (arg: any) => void;
 };
 
 function createEvents<F extends Function>(): Events<F> {
-  let handlers: F[] = [];
+  let handlers = new Set<F>();
```

Comment:
Oh, also I don't know what kind of browser support you want, but I replaced the `createEvents` implementation with one that uses a `Set` which has limited support in IE 11 (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set#browser_compatibility)